torch/distributed/checkpoint/filesystem.py

```diff
@@ -63,6 +68,10 @@ class _StoragePrefix:
 DEFAULT_SUFFIX = ".distcp"
 
 
+def _generate_uuid() -> str:
+    return str(uuid.uuid4())
```

Comment:
If we generate the save_id with uuid, all ranks will have different save_id. Is this acceptable? Can this confuse users if we use the save_id for logging/debugging purpose

Reply:
Turns out only rank0 will save this ID to the checkpoint since the metadata object is managed by this object.